gensim/models/keyedvectors.py

```diff
@@ -1017,7 +1028,9 @@ def _log_evaluate_word_analogies(section):
             logger.info("%s: %.1f%% (%i/%i)", section['section'], 100.0 * score, correct, correct + incorrect)
             return score
 
-    def evaluate_word_analogies(self, analogies, restrict_vocab=300000, case_insensitive=True, dummy4unknown=False):
+    def evaluate_word_analogies(
```

Comment:
Please add some unit tests to cover the newly added functionality.

Reply:
Done